boto3/session.py

```diff
@@ -174,14 +175,21 @@ def client(self, service_name, region_name=None, api_version=None,
         :param aws_session_token: The session token to use when creating
             the client.  Same semantics as aws_access_key_id above.
 
+        :type config: botocore.client.Config
+        :param config: Advanced client configuration options. If region_name
+            is specified in the client config, its value will take precedence
+            over environment variables and configuration values, but not over
+            a region_name value passed explicitly to the method.
+
         :return: Service client instance
+
         """
         return self._session.create_client(
             service_name, region_name=region_name, api_version=api_version,
             use_ssl=use_ssl, verify=verify, endpoint_url=endpoint_url,
             aws_access_key_id=aws_access_key_id,
             aws_secret_access_key=aws_secret_access_key,
-            aws_session_token=aws_session_token)
+            aws_session_token=aws_session_token, config=config)
 
     def resource(self, service_name, region_name=None, api_version=None,
```

Comment:
I also like the latter of the options you listed.

If we do the first option, I feel that we should apply ``'Resources'`` to the beginning or end of whatever they provide. Probably to the beginning to make it easier to determine from the user agent if a resource is being used.

We will probably get asked this as a feature request. I feel that we can wait till we get a request for either of the two to a get better sense of what users prefer. Either change should be additive in the end.